Fix bytes vs. string issue under Py3

hornc opened the pull request:
<!-- What issue does this PR close? -->
From discussion in #2854 
Hopefull fixes the Python3 test error :crossed_fingers: 

Python2
```
>>> import zlib
>>> zlib.compressobj(9).compress('hello')
'x\xda'
```
AND
```
>>> zlib.compressobj(9).compress(b'hello')
'x\xda'
```


Python3:
```
>>> import zlib
>>> zlib.compressobj(9).compress('hello')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: a bytes-like object is required, not 'str'
```

And successfully:

```
>>> zlib.compressobj(9).compress(b'hello')
b'x\xda'
```



<!-- What does this PR achieve? [feature|hotfix|fix|refactor] -->

### Technical
<!-- What should be noted about the implementation? -->

### Testing
<!-- Steps for reviewer to reproduce/verify what this PR does/fixes. -->

### Evidence
<!-- If this PR touches UI, please post evidence (screenshots) of it behaving correctly. -->

### Stakeholders
<!-- @ tag stakeholders of this bug -->

cclauss commented:
This makes these particular test pass but is it what we really want?

Don't we want to be able to compress and decompress structures that contain text strings?

hornc commented:
@cclauss I think you are correct, I did this quickly and thought it was just the seed that needed to be bytes. I will close this and review your #2862 in preference. I think the test I deleted should be moved to its own test file to protect against the error I introduced here, and we can have it run at the appropriate time too.
